### ISSUE TITLE
Reset button fails if TextInput value is set by defaultValue property

### ISSUE THREAD

**Issue by JeremyColton** [Albert-Gao/veasy#24]
In the `<TextInput>` I set the value via the `defaultValue` property as advised via this bug's workaround:
https://github.com/Albert-Gao/veasy/issues/18

But, this causes the "reset" button to not reset the form's values!

**Comment by Albert-Gao** [comment]
Yes, sounds like a un-covered case for me. it's a bug, i will add a test case for this and fix it asap. thanks!

**Comment by JeremyColton** [comment]
Very much looking forward to this fix 🥇

**Comment by Albert-Gao** [comment]
Found a dirty fix for forcing it update if you use `defaultValue` to bind the data:

Just add a `key={value ? 'isNew' : 'MaybeOld'}` to the inside component, like the `<input>` inside `<TextInput>`

A minor bug is it will trigger the validation and lose focus when first check.
But the reset function now works as expected.

**Comment by JeremyColton** [comment]
Thanks for the update.  You mean to add the `key` like this:

```const TextInput = ({
  name,
  placeholder,
  status,
  errorText,
  value,
  onChange
}) => (
  <div>
    <label htmlFor={name}>
      <input
        type="text"
        id={name}
        name={name}
        className={status}
        placeholder={placeholder}
        defaultValue={value}
        onChange={onChange}
        key={value ? 'isNew' : 'MaybeOld'}
      />
      <p>{errorText}</p>
    </label>
  </div>
);

TextInput.propTypes = {
  name: PropTypes.string.isRequired
};```

**Comment by Albert-Gao** [comment]
Yes

**Comment by JeremyColton** [comment]
I can type a letter and then as you said, the field loses focus.  Not a good user experience. It's amazing the 'text jumping' bug is a feature of React! I don't see this in Facebook and I'm guessing it's all written in React.  They use the 'ref' workaround?

**Comment by Albert-Gao** [comment]
Haha, I wouldn't say it's a feature. But it is indeed an expected behavior of a controlled component. I didn't look into the workaround. You need to find it for yourself. But Dan indeed mentioned something like `ref`. That is the uncontrolled solution. Maybe it is not compatible with `veasy` considered that `veasy` is fully controlled component. :)

**Comment by Albert-Gao** [comment]
Sorry for the trouble, you mean the reset button won't work when you set the value of the field from `defaultValue`?

I am gonna have a try and fix this.

**Comment by Albert-Gao** [comment]
The solution is here.

https://github.com/Albert-Gao/veasy/issues/18#issuecomment-380993303

I think you close this issue because you get it right, the root of cause is not the `defaultValue`, it is simply shouldn't be used in this case.

So check that comment. Hope everything works well now. :)

**Comment by JeremyColton** [comment]
Yes, your fix in #18 is great! Works perfectly!
